### DISCUSSION TITLE
如何在umijs/max中同时使用vue2和react

### DISCUSSION THREAD

**Discussion by ghost** [umijs/umi#10106] [Q&A]
启动了umijs/max项目，同时配置了vue-loader等 但是运行后错误信息为
<img width="525" alt="image" src="https://user-images.githubusercontent.com/71827583/208379184-23ab8084-d8f9-443b-a036-da2e5257e215.png">
 
package.json
 "dependencies": {
    "@ant-design/icons": "^4.7.0",
    "@ant-design/pro-components": "^2.0.1",
    "@umijs/max": "^4.0.36",
    "antd": "^5.0.0",
    "vue": "^2.7.14",
    "vuera": "^0.2.7"
  },
  "devDependencies": {
    "@types/react": "^18.0.0",
    "@types/react-dom": "^18.0.0",
    "file-loader": "^6.2.0",
    "husky": "^8.0.1",
    "lint-staged": "^13.0.3",
    "prettier": "^2.7.1",
    "prettier-plugin-organize-imports": "^2",
    "prettier-plugin-packagejson": "^2",
    "style-loader": "^3.3.1",
    "typescript": "^4.1.2",
    "url-loader": "^4.1.1",
    "vue-loader": "^15.10.1",
    "vue-template-compiler": "^2.7.14",
    "worker-loader": "^3.0.8"
  }

**Comment by fz6m** [comment]
需要取决于你的 vue 部分以什么方式进入项目，react 项目是不支持 vue 享受约定式路由等特性的。

如果你只是想把 vue 渲染到某个节点上，通过配置应该是可以实现的。

另外，umi 4 现在支持纯 vue 项目，可以创建一个新的 vue 模板项目进行 vue 的开发。

不推荐项目中混用 react + vue ，这在 umi 这种脚手架中是很难做到的，如果确有此需求，更好的方式是自己配置一个原生的 webpack 项目。
